final/src/main/java/ua/epam/spring/hometask/dao/EventDAO.java

```diff
@@ -0,0 +1,205 @@
+    }
+
+    /**
+     * Checks if event airs on particular date
+     *
+     * @param airDates air dates of event
+     * @param date     Date to ckeck
+     * @return <code>true</code> event airs on that date
+     */
+    default boolean airsOnDate(NavigableSet<LocalDateTime> airDates, LocalDate date) {
+        return airDates.stream().anyMatch(dt -> dt.toLocalDate().equals(date));
+    }
+
+    /**
+     * Checks if event airs on particular date and time
+     *
+     * @param airDates air dates of event
+     * @param dateTime Date and time to check
+     * @return <code>true</code> event airs on that date and time
+     */
+    default boolean airsOnDateTime(NavigableSet<LocalDateTime> airDates, LocalDateTime dateTime) {
+        return airDates.stream().anyMatch(dt -> dt.equals(dateTime));
+    }
+
+    /**
+     * Checking if datetime in range.
+     *
+     * @param current Current datetime to check
+     * @param from    Start datetime to check
+     * @param to      End datetime to check
+     * @return <code>true</code> datetime in range
+     */
+    default boolean dateTimeInRange(LocalDateTime current, LocalDateTime from, LocalDateTime to) {
+        return current.compareTo(from) >= 0 && current.compareTo(to) <= 0;
+    }
+
+    /**
+     * Checking if date in range.
+     *
+     * @param current Current date to check
+     * @param from    Start date to check
+     * @param to      End date to check
+     * @return <code>true</code> date in range
+     */
+    default boolean dateInRange(LocalDate current, LocalDate from, LocalDate to) {
+        return current.compareTo(from) >= 0 && current.compareTo(to) <= 0;
+    }
+
+    /**
+     * Get auditorium on particular <code>dateTime</code>
+     *
+     * @param auditoriums auditoriums for event
+     * @param dateTime    Date and time of aired event
+     * @return auditorium on particular <code>dateTime</code>
+     */
+    default Auditorium findAuditoriumOnDateTime(NavigableMap<LocalDateTime, Auditorium> auditoriums, LocalDateTime dateTime) {
+        return auditoriums.get(dateTime);
+    }
+
+    /**
+     * Checks if event is aired on particular <code>dateTime</code> and assigns
+     * auditorium to it.
+     *
+     * @param airDates    air dates of event
+     * @param auditoriums auditoriums for event
+     * @param dateTime    Date and time of aired event for which to assign
+     * @param auditorium  Auditorium that should be assigned
+     * @return <code>true</code> if successful, <code>false</code> if event is
+     * not aired on that date
+     */
+    default boolean assignAuditorium(NavigableSet<LocalDateTime> airDates, NavigableMap<LocalDateTime, Auditorium> auditoriums, LocalDateTime dateTime, Auditorium auditorium) {
+        if (airDates.contains(dateTime)) {
+            auditoriums.put(dateTime, auditorium);
+            return true;
+        } else {
+            return false;
+        }
+    }
```

Comment:
1) Is it necessary to use default method here? You can use save method for updating event and all other logic move to service (adding new auditorium to the list).
2) Instead of NavigableSet<LocalDateTime> airDates, NavigableMap<LocalDateTime, Auditorium> auditoriums you can pass Event event
3) Extra else condition, you can use directly return false;